### DISCUSSION TITLE
Custom Fonts for the Paint Editor and Animated Text extension

### DISCUSSION THREAD

**Discussion by BlueDome77** [ScratchAddons/ScratchAddons#6164] [Ideas] [answered]
Allowing people to upload their own fonts (.otf .ttf etc) into the paint editor or the Animated Text extension would be a game-changer for people who make games in Scratch or Turbowarp!
![Screenshot 2023-06-07 at 11 44 12 AM](https://github.com/ScratchAddons/ScratchAddons/assets/135789443/e57f41bd-4f72-4e4e-8ccb-e1bce8249677)

**Comment by CST1229** [comment]
I'm pretty sure Scratch literally doesn't let you use uploaded fonts in SVGs (only ones on your computer, through name). Stuff related to the Animated Text extension in TurboWarp goes in [TurboWarp/extensions](https://github.com/TurboWarp/extensions).

**Reply by BlueDome77** [reply]
Yes it does: https://meeeeeeeep1.github.io/haven.scratch-tools/customScratchFont/

**Reply by CST1229** [reply]
> Yes it does: [meeeeeeeep1.github.io/haven.scratch-tools/customScratchFont](https://meeeeeeeep1.github.io/haven.scratch-tools/customScratchFont/)

That doesn't let you upload a .ttf or .otf file.

**Reply by BlueDome77** [reply]
It doesn't but I was just proving to you that Scratch **does** let you use uploaded fonts in the paint editor

**Reply by CST1229** [reply]
> It doesn't but I was just proving to you that Scratch **does** let you use uploaded fonts in the paint editor

Apparently you can use a data URL as a font in an SVG in Scratch, but editing it in the paint editor removes the font.

**Reply by BlueDome77** [reply]
**Web Browser:** Safari Version 16.5
**Scratch Version:** Vanilla
**Userscripts:** None

https://github.com/ScratchAddons/ScratchAddons/assets/135789443/72b62cef-6891-458b-8b02-eab426fc568c

**Reply by CST1229** [reply, marked as answer]
That is for a font that is already installed on your own computer. I'm talking about a font that is **not** on your computer, and is embedded into the SVG. Try with this one (right click it -> save image as, then import into Scratch):
![font](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/9a51cea9-2ef1-401a-b6c4-2779a0a2db53)

<br />
<br />

The result is:
![image](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/5be45a3c-13d9-4a89-9f3d-67077a073bf5)
![image](https://github.com/ScratchAddons/ScratchAddons/assets/68464103/d403c08a-8b7b-4e5b-b3b5-57ae0ff8de81)
